addons/implied-repos/common/src/main/java/org/commonjava/indy/implrepo/change/ImpliedRepositoryDetector.java

```diff
@@ -399,19 +400,29 @@ else if ( !config.isIncludeSnapshotRepos() && !repo.isReleasesEnabled() )
                             e );
                 }
 
+                final RemoteRepository ref = creator.createFrom( gav, repo, LoggerFactory.getLogger( creator.getClass() ) );
+                if ( ref == null )
+                {
+                    logger.warn(
+                            "ImpliedRepositoryCreator didn't create anything for repo: {}, specified in: {}. Skipping.",
+                            repo.getId(), gav );
+                    continue;
+                }
+
+                if ( rrs != null && !rrs.isEmpty() )
+                {
+                    rrs = rrs.stream()
+                             .filter( rr -> rr.isAllowReleases() == repo.isReleasesEnabled() )
+                             .filter( rr -> rr.isAllowSnapshots() == repo.isSnapshotsEnabled() )
+                             .filter( rr -> rr.getPathMaskPatterns() == null || rr.getPathMaskPatterns().isEmpty() )
```

Comment:
Also here I would tend to compare the value with the one in ref since we already have it available here.

Reply:
Yeah, it's better to add that condition too,  to avoid to create a duplicate one with the same pathMaskPatterns.